### PR TITLE
BatchNorm: Document CUDNN_BN_MIN_EPSILON

### PR DESCRIPTION
**Pull request by leezu**
When using an eps value <= CUDNN_BN_MIN_EPSILON cudnn will fail with CUDNN_STATUS_BAD_PARAM.

### CODE REVIEW

**Comment by sxjscience** [comment]
Also check the parameter here https://github.com/dmlc/mxnet/blob/master/src/operator/cudnn_batch_norm-inl.h#L29-L37.

**Comment by leezu** [comment]
I added a check comparing "param.eps > CUDNN_BN_MIN_EPSILON"

**Comment by leezu** [comment]
I now `#include <cudnn.h>` to get access to `CUDNN_BN_MIN_EPSILON`. Please let me know if you prefer not to include and instead to compare to constant 1e-5.

File: src/operator/cudnn_batch_norm-inl.h [hunk 1]
```diff
@@ -13,6 +13,10 @@
 #include <utility>
 #include "./batch_norm-inl.h"
 
+#if MXNET_USE_CUDNN == 1
+#include <cudnn.h>
+#endif  // MXNET_USE_CUDNN == 1
```
Comment (sxjscience) [inline src/operator/cudnn_batch_norm-inl.h]: These lines may not be necessary, we can directly call CUDNN_BN_MIN_EPSILON

**Comment by leezu** [comment]
Thanks. I removed the unneeded import.

**Comment by piiswrong** [comment]
ok

**Comment by sxjscience** [comment]
Let's merge?

**Comment by leezu** [comment]
I don't see any reason not to, so I think you can go forward to merge.

**Pull request merged**